### PR TITLE
Feature/server/169 scenario management

### PR DESCRIPTION
**Pull request by YIllmann**

### CODE REVIEW

**Comment by simohlsen** [comment]
Would be great to make the changes to the database merge-able asap so new issues can be implemented on the latest state of the database and wont have to be potentially migrated.

**Comment by YIllmann** [comment]
First of: sry for the lasting PR. The changes that were caused of this PR are massive. Thank you for your help an patience. Since your two PRs the following changes were made and require further review:
- added files in web_api: all files are very lookalike due to similar requirements. possible updates due FE changes might follow outside of this PR.
- models: It appears that we both missed the references for vehicles to location in our PlayersToVehicle Table. Therefore i added a column and extended the Unique Constraint to ensure valid data. Luckily the test-data you created represent a special case of mapping :).
- test_action only contains a little test i used with my debugger to ensure the media is updated correctly
- test_scenario does the same for our execution-scenario-vehicle problem.

File: server/scenario/web_api/data/action.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    ).filter(
+        models.ResourcesNeeded.action_id == action_id
+    ).all()
+
+    return {
+        "id": action.id,
+        "name": action.name,
+        "min_role": min_role.name if min_role else None,
+        "duration_secs": action.duration_secs,
+        "media_refs": json.loads(action.media_refs) if action.media_refs else [{}],
```
Comment (simohlsen) [inline server/scenario/web_api/data/action.py:56]: Why `[{}]` not `[]`?

File: server/scenario/web_api/data/action.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    """ Creates an empty Action and returns the dbo as JSON. """
+    action = models.Action(
+        name="Neue Maßnahmen",
+        duration_secs=0,
+        required_power=0,
+        media_refs=[{}],
+        result=""
+    )
+    db.session.add(action)
+    db.session.commit()
```
Comment (simohlsen) [inline server/scenario/web_api/data/action.py:78]: What if you hit the new action button twice without editing the first or is this not possible?

File: server/scenario/web_api/data/location.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            } for location_relation in location_relations
+        ]
+    }
+
+
+@web_api.post("/location")
+def create_location():
+    location = models.Location(
+        name="Neue Location",
+        media="[{}]",
```
Comment (simohlsen) [inline server/scenario/web_api/data/location.py:64]: Why `[{}]` not `[]` or just `null`? Also, shouldn't this be `media_refs`?

**Review by simohlsen — CHANGES_REQUESTED** [review]

File: server/scenario/web_api/data/action.py (hunk 2 above)
Comment (YIllmann) [reply server/scenario/web_api/data/action.py:78]: the action id autoincrements and you get 2 action with the same name displayed. no collision on db-side. further, you will be redirected in a different edit window, so you cant create 2 actions. Locations and resource behave the same

File: server/scenario/web_api/data/resource.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+
+@web_api.post("/resource")
+def create_resource():
+    resource = models.Resource(
+        name="Neue Resource",
+        consumable="False",
+        quantity=10000,
+        media_refs="[{}]"
```
Comment (simohlsen) [inline server/scenario/web_api/data/resource.py:55]: `[]` or `null`

**Review by simohlsen — CHANGES_REQUESTED** [review]

**Review by simohlsen — APPROVED** [review]

**Pull request merged**